players/interface_player.py

```diff
@@ -0,0 +1,17 @@
+class IPlayer:
```

Comment:
Why is this an interface? You only have 1 implementation.
Also, this kind of interfacing is very c++, not python :(

Reply:
fixed, got a link to read about python interfaces?

Reply:
Fixed, have an example\link for python interface?

Reply:
This is the how:
https://docs.python.org/3/library/abc.html
This is the why:
https://www.python.org/dev/peps/pep-3119/

Reply:
I started with abc and moved away because I thought it was only for python 3+.